Update User Specification (#12)
UCLCSSA/User-Specification: Function-Specification.md

# UCLCSSA WeChat App Functional Specification

## Terminology

This specification uses the terms **MUST, SHOULD, MAY, MUST NOT, MAY NOT, SHOULD

> NOT, RECOMMENDED, OPTIONAL, REQUIRED** as defined in [RFC2119](https://tools.ietf.org/html/rfc2119).

Reviewer: ```suggestion
NOT, RECOMMENDED, OPTIONAL, REQUIRED, SHALL** as defined in [RFC2119](https://tools.ietf.org/html/rfc2119).
```

| **LOCAL TIMEZONE** | The timezone where UCL is located. Note the the timezone needs to take into account Daylight Savings adjustments. |
| **LOCAL CITY**     | London.                                                                                                           |
| **CSSA**           | UCL Chinese Students and Scholars Association.                                                                    |
| **ADMIN**          | Administrator                                                                                                     |
| **CSSA-CARD**      | Discount card published by UCL CSSA.                                                                              | 

## Functional Requirements

### 1. Main Page

![Main Page Screenshot 1](img/main-page-1.png)

#### 1.1 Weather and Date

> This part should contain two parts Date and the Weather. We use LOCAL TIMEZONE and LOCAL CITY.

Reviewer: ```suggestion
This part should contain:
1. Date. Use LOCAL TIMEZONE.
2. Weather. Use weather for LOCAL CITY.
```

   4. Tutorial
   5. Exam (Dependent on UCL API Functionalities)

> > TODO: Does UCL Timetable include exam timetables?   

Reviewer: If I recall correctly UCL API does not include exam timetables. Exam timetable related content should be removed.

The user SHOULD be able to choose his/her favorite study facility (i.e. pin it).

##### 1.4.5 Tapping progress bar

Jump to library seats page.

> TODO: Jump to library seats page.
> TODO: Pending Design.

### 2. Post Page  

![Post Page Screenshot 1](img/post-page-1.png)

#### 2.1 Post Horizontal Slider  

User MAY post information of different categories.  

> This MUST includes:  

Reviewer: ```suggestion
The categories MUST include:
```

> - 房屋交友  

Reviewer: Does housing really go well together with making friends? Maybe this should be separate categories,
- Housing
- Socializing (?)

Reviewer 2: Definitely not Socializing. My original thought was “房屋室友”, but maybe “房屋出租” would be better? This is trivial though.

Reviewer: Oh that makes sense. Since here it is written "房屋交友" I thought it was Housing + Making Friends. Renting would make sense.

Reviewer: I think "房屋出租" would probably be better for such renting use case.

> - 房屋交友  

Reviewer: ```suggestion
- 房屋出租
```

Change this to better reflect the renting use case.

- 二手闲置  
- 提问回答  
- 失物招领  

##### 2.1.1 Interact with user  

- It is REQUIRED that the horizontal slider only moves when the user scrolls it.

- When user press specific section, that section should be highlighted as shown in the design and the posts displayed should be filtered.

#### 2.2 Post Block

> Each *Post* inside post page will be displayed as a post block. A post block should include following elements.

Reviewer: ```suggestion
Each *post* inside each category page SHALL be displayed as a post block. A *post block* should include the following elements:
```

- Time
- Title

> - Brief content

Reviewer: I'm assuming this means a limited number of characters extracted from the post's content?

Reviewer 2: > I'm assuming this means a limited number of characters extracted from the post's content?

I guess if the post is too long, only the first part of the post will be displayed on the post feed? and the user can click on the block to see the full content.

Reviewer: Makes sense. Say you have a block

> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum vestibulum, dolor at rhoncus molestie, diam leo pellentesque velit, in molestie dui ipsum ac quam. Mauris ac nisi ligula. Nulla dapibus quam eu nisi aliquam commodo. In cursus lorem eu velit elementum pretium. Aenean sagittis metus gravida ipsum feugiat, id venenatis sapien blandit. Fusce quis malesuada nunc. Mauris risus dolor, ultrices vel eleifend in, bibendum volutpat nulla. Vestibulum vel ante et nisi tempus posuere. Vestibulum quis mauris ultrices lectus sagittis mattis id ac mauris. Nunc euismod, erat eget cursus volutpat, sapien magna gravida velit, at mollis nisl tellus sit amet nisl. In ac nisi turpis. Cras ut vestibulum nunc, vel elementum metus. 

Only the first `x` characters would be shown, as in

> Lorem ipsum dolor sit amet, consectetur...
> **Click to view post in detail**

- User Information
- Photo

##### 2.2.1 Time  

LOCAL TIMEZONE  

##### 2.2.2 Title  

>TODO: Discuss how many characters *Title* should include?  

##### 2.2.3 Brief Content  

>TODO: Discuss how many characters *Brief Content* should include?  

##### 2.2.4 Photo  

> No more than 3 photos.  

Reviewer: Further details regarding photo size limits will need to be determined.

##### 2.2.5 User Information  

User Profile and ID.  

#### 2.3 Post Button[WECHAT LOGIN REQUIRED]

>TODO: Pending design.    

### 3. Restaurant Page  

![Restaurant Page Screenshot 1](img/restaurant-page-1.png)

#### 3.1 CSSA CARD(Wechat Login)

When pressed, display the CSSA-CARD.  

#### 3.2 Restaurant Block  

Restaurant Block should include following elements  

- Name of restaurant

> - Photo of restauran

Reviewer: ```suggestion
- Photo of restaurant
```
